plugins/parcelize/parcelize-compiler/src/org/jetbrains/kotlin/parcelize/ir/irUtils.kt

```diff
@@ -32,9 +33,15 @@ import org.jetbrains.kotlin.parcelize.serializers.ParcelizeExtensionBase
 import org.jetbrains.kotlin.parcelize.serializers.ParcelizeExtensionBase.Companion.CREATOR_NAME
 import org.jetbrains.kotlin.types.Variance
 
+private val IrClass.sealedSuperClass: IrClass?
+    get() = superTypes.mapNotNull { superType ->
+        superType.classOrNull?.owner?.takeIf { it.modality == Modality.SEALED }
+    }.singleOrNull()
+
 // true if the class should be processed by the parcelize plugin
 val IrClass.isParcelize: Boolean
-    get() = kind in ParcelizeExtensionBase.ALLOWED_CLASS_KINDS && hasAnyAnnotation(PARCELIZE_CLASS_FQ_NAMES)
+    get() = kind in ParcelizeExtensionBase.ALLOWED_CLASS_KINDS &&
+            (hasAnyAnnotation(PARCELIZE_CLASS_FQ_NAMES) || sealedSuperClass?.hasAnyAnnotation(PARCELIZE_CLASS_FQ_NAMES) == true)
```

Comment:
Sealed interfaces are possible since 1.5, so a class can have several sealed supertypes:
```
sealed class A
sealed interface I

class C : A(), I
```

Reply:
Completely missed that! Changed the check to consider all supertypes and to allow the `Parcelize` annotation on sealed interfaces. I've also added a test for sealed Parcelable interfaces in `plugins/parcelize/parcelize-compiler/testData/box/sealedInterface.kt`.